packages/kbn-health-gateway-server/src/kibana/routes/status.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import https from 'https';
+import type { Request, ResponseToolkit } from '@hapi/hapi';
+import nodeFetch, { RequestInit, Response } from 'node-fetch';
+import type { IConfigService } from '@kbn/config';
+import type { Logger } from '@kbn/logging';
+import type { KibanaConfigType } from '../kibana_config';
+import { KibanaConfig } from '../kibana_config';
+
+const GATEWAY_STATUS_ROUTE = '/api/status';
+const KIBANA_STATUS_ROUTE = '/api/status';
+
+interface StatusRouteDependencies {
+  log: Logger;
+  config: IConfigService;
+}
+
+export function createStatusRoute({ config, log }: StatusRouteDependencies) {
+  const kibanaConfig = new KibanaConfig(config.atPathSync<KibanaConfigType>('kibana'));
+
+  return {
+    method: 'GET',
+    path: GATEWAY_STATUS_ROUTE,
+    handler: async (req: Request, h: ResponseToolkit) => {
+      const responses = await fetchKibanaStatuses(kibanaConfig, { log });
+      const { body, statusCode } = mergeStatusResponses(responses);
+      return h.response(body).type('application/json').code(statusCode);
+    },
+  };
+}
+
+async function fetchKibanaStatuses(kibanaConfig: KibanaConfig, { log }: { log: Logger }) {
+  const fetch = configureFetch(kibanaConfig);
```

Comment:
I don't think we need to call `configureFetch` (or at least the underlying `generateAgentConfig` call) for each call to the proxy's route.

Reply:
good call, will update